crates/bevy_ecs/src/storage/sparse_set.rs

```diff
@@ -28,59 +30,87 @@ impl<I, V> SparseArray<I, V> {
 }
 
 impl<I: SparseSetIndex, V> SparseArray<I, V> {
+    fn split_index(index: I) -> (usize, usize) {
+        let idx = index.sparse_set_index();
+        (idx / PAGE_SIZE, idx % PAGE_SIZE)
+    }
+
+    fn make_page() -> Box<[Option<V>; PAGE_SIZE]> {
+        // SAFE: The memory is all initialized to None upon return.
+        unsafe {
+            let mut page: MaybeUninit<[Option<V>; PAGE_SIZE]> = MaybeUninit::uninit();
+            let array = page.as_mut_ptr().as_mut().unwrap();
+            for item in &mut array[..] {
+                *item = None;
+            }
+            // TODO: Initialize with Box::assume_uninit when https://github.com/rust-lang/rust/issues/63291 lands in stable.
+            Box::new(page.assume_init())
```

Comment:
I'm pretty sure this is unsound. The `as_mut` will create a `&mut Option<V>` which is not initialized, which is insta-UB. The `MaybeUninit` docs show an example of how to initialize an array with `ptr.write`: https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element